src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier{
+    Random random = new Random();
+    ColorSupplier colorSupplier = new ColorSupplier();
+    public Figure getRandomFigure(Figure[] figures){
+        int index = random.nextInt(figures.length);
+        if(index == 0){
+            Figure circle = new Circle(random.nextDouble(20), colorSupplier.getRandomColor());
+            return circle;
+        } else if(index == 1){
+            Figure rectangle = new Rectangle(random.nextDouble(20), random.nextDouble(20),
+                    colorSupplier.getRandomColor());
+            return rectangle;
+        } else if (index == 2){
+            Figure rightTriangle = new RightTriangle(random.nextDouble(20), colorSupplier.getRandomColor());
+            return rightTriangle;
+        } else if (index == 3) {
+            Figure isoscelesTrapezoid = new IsoscelesTrapezoid(random.nextDouble(20),
+                    random.nextDouble(20), random.nextDouble(20), colorSupplier.getRandomColor());
```

Comment:
Avoid passing expressions directly in the constructor. Instead, assign them to variables for better readability and maintainability.